{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# New things I learned from the AMIA 2018 informatics summit \n",
    "\n",
    "\n",
    "\n",
    "## _Zichen Wang_    \n",
    "## Ma'ayan Lab Meeting @ 04/13/2018\n",
    "---\n",
    "\n",
    "\n",
    "AMIA: American Medical Informatics Association\n",
    "\n",
    "Major areas: \n",
    "\n",
    "0. Translational Bioinformatics (TBI)\n",
    "0. Clinical Research Informatics (CRI)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "S40: Panel - Deep Learning for Healthcare - Hype or the real thing?\n",
    "---\n",
    "\n",
    "slides are available on http://sunlab.org/\n",
    "\n",
    "# Doctor AI\n",
    "## Jimeng Sun @ Georgia Tech\n",
    "\n",
    "computational phenotyping from EHR\n",
    "Deep learning for phenotyping\n",
    "- targets: dz risk cost, outcome\n",
    "- intermediate layers > phenotypes?\n",
    "![](img/DL_for_phenotyping.png)\n",
    "\n",
    "performance and interoperability\n",
    "- Heart Failure RNN: Predict future onset of HF given longitudinal EHR\n",
    "- Dr.AI: Temporal modeling of longitudinal EHR\n",
    "- Med2Vec: medical concept representation learning\n",
    "- RETAIN, CAML: automated ICD coding from clinical notes\n",
    "- medGAN: generate synthetic EHR\n",
    "\n",
    "\n",
    "# DL evidence-based medicine?\n",
    "## Marzyeh Ghassemi @MIT & Verily\n",
    "\n",
    "LipNet: lipreading\n",
    "![](img/lipreading.png)\n",
    "\n",
    "Clinical event prediction\n",
    "\n",
    "- attention model: vital signs vs time, notes, [paper](https://arxiv.org/pdf/1705.08498.pdf)\n",
    "    - Real-time prediction of next clinical intervention \n",
    "\n",
    "![](img/Clinical_event_preds-1.png)\n",
    "![](img/Clinical_event_preds-2.png)\n",
    "\n",
    "- **Reinforcement learning** to learn optimal treatment for sepsis\n",
    "- **Semi-supervised** (mixture of paired and unpaired training samples) biomedical translation with Cycle Wasserstein Regression, [paper](http://www.marzyehghassemi.com/wp-content/uploads/2018/01/semi-supervised-CWR-GAN_Ghassemi.pdf)\n",
    "    0. Individualized Treatment Effect Prediction using EMR\n",
    "    0.  Transcriptomics Extrapolation using L1000\n",
    "\n",
    "![](img/CWR-GAN.png)\n",
    "\n",
    "- Digital phenotyping (outside of clinics)\n",
    "\t+ active data (surveys)\n",
    "\t+ passive: wearables\n",
    "- Verily Mood Study: sleep, physical activity, etc.\n",
    "\n",
    "\n",
    "# DL in clinical NLP\n",
    "## Hong Yu @ UMass\n",
    "\n",
    "MADE1.0 clinical NLP challenge    \n",
    "at least 70% ADEs do not appear in ICD\n",
    "\n",
    "Challenges:\n",
    "- noisy/incomplete sentence structure\n",
    "- inconsistent syntactic cues\n",
    "- medical jargons/abbreviations\n",
    "- multi-word entities\n",
    "\n",
    "Med events:\n",
    "- indication\n",
    "- ADE\n",
    "- symptom / dz\n",
    "- meds\n",
    "\n",
    "Detecting ADEs CNN+RNN autoencoder\n",
    "Relation identification: SVM outperforms DL\n",
    "\n",
    "# Current limit of DL in health care\n",
    "## David Sontag\n",
    "\n",
    "DL imaging classification in Patho, Radiology, arrhythmia\n",
    "\n",
    "Challenges:\n",
    "- class imbalance\n",
    "- 3+ D inputs\n",
    "- object detection: scale variance, large images where fine detail is important\n",
    "- datasets too small\n",
    "- not obvious extend to:\n",
    "\t+ multi-modal data\n",
    "\t+ context specific: age/comobidity\n",
    "\t+ predict treatment effect\n",
    "\t+ integrate physician knowledge\n",
    "\n",
    "Risk stratification on longitudinal clinical data\n",
    "\n",
    "baseline model feature vector (42K dims)\n",
    "- all history\n",
    "- 24 months\n",
    "- ...\n",
    "\n",
    "RNNs (next word prediction) do not significantly outperform baseline\n",
    "- sequential data in medicine is very different from language\n",
    "\n",
    "![](img/baseline_model.png)\n",
    "\n",
    "\n",
    "# Tasks automatable by DL\n",
    "## M. Brandon Westover @ MGH\n",
    "\n",
    "- sleep scoring\n",
    "CNN+RNN for sleep stage classification\n",
    "![](img/sleep_scoring.png)\n",
    "\n",
    "- seizure detection\n",
    "30TB EEG files\n",
    "co-learning of human and machine\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Short talks\n",
    "---\n",
    "\n",
    "# Personalized Vital Sign Modeling to Predict Health Status\n",
    "## Jessilyn Dunn (Mike Snyder lab)\n",
    "\n",
    "- [iPOP (Integrated Personal Omics Profiling) study](http://snyderlab.stanford.edu/iPOP.html)\n",
    "\t+ integrate omics with wearable sensors (skin temp, heart rates and etc.) + clinical vitals\n",
    "\t+ T2D risks\n",
    "\t+ personalized interventions\n",
    "\t+ 100 individuals + 4 years (capture environ changes: viral infection/diet change/stress)\n",
    "\t+ watch data \n",
    "\t\t- heart rate, skin temp, accelermetry, circadian diff\n",
    "\t\t- _PLOS Biology_ 2017: Li, Dunn, Salins _et al_\n",
    "\n",
    "- clinical utilities of wearable sensors \n",
    "\t+ Change-of-Heart algorithm: detect abnormal events for individuals \n",
    "\t+ picked up Lyme and viral infections, high CRP (inflammation indicator)\n",
    "\t+ physiological diff in insulin resistance/sensitivity\n",
    "\n",
    "- clinical vital signs\n",
    "\t+ Stanford EMR: 28K patients, 31M lab tests, 855K vitals\n",
    "\t+ 50 tests from blood/urine tests\n",
    "\t+ Use pulse/temp to predict blood/urine tests: linear mixed model for individual baselines\n",
    "\n",
    "- personalized vitals (**clinical + wearable**): \n",
    "Can remotely collected vitals predict clinical labs?\n",
    "\n",
    "\n",
    "--- \n",
    "\n",
    "\n",
    "# Generative Adversarial Networks: Evaluating Drug Effect in Synthetic Time Series Generated from EHR Data\n",
    "## Alexandre Yahi @ Nick Tatonetti lab\n",
    "\n",
    "Synthetic data generation to protect patient privacy.\n",
    "\n",
    "SYNTHEA\n",
    "\n",
    "DLE: drug laboratory effects\n",
    "Statin > cholesterol level\n",
    "4.8K patients\n",
    "\n",
    "- totalGAN\n",
    "- subGANs (more accurate):\n",
    "ICD9 + ATC > 1.5K features > AE > 32 features > tSNE > 3 clusters of patients\n",
    "\n",
    "# Tandem Rank-Goodness of Fit is a Feasible Measure to Guide Agnostic Metabolic Profile Selection in Automated Machine Learning Applications\n",
    "## Matthew Breitenstein\n",
    "\n",
    "Metformin treatments vs controls\n",
    "42 metabolites\n",
    "BMI higher in Metformin cases \n",
    "\n",
    "TPOT: **automated ML** tool\n",
    "![](https://github.com/EpistasisLab/tpot/raw/master/images/tpot-demo.gif)\n",
    "\n",
    "TPOT will automate the most tedious part of machine learning by intelligently exploring thousands of possible pipelines to find the best one for your data.\n",
    "![](https://raw.githubusercontent.com/EpistasisLab/tpot/master/images/tpot-ml-pipeline.png)\n",
    "\n",
    "\n",
    "# Characterizing drug-related adverse events by joint analysis of biomedical and genomic data: A case study of drug-induced pulmonary fibrosis\n",
    "## Alex Jiang\n",
    "\n",
    "FAERS [AERSMine](https://research.cchmc.org/aers/)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Posters\n",
    "---\n",
    "\n",
    "- Systematic detection of height, weight outliers in EHR:    \n",
    "    + IBM Watson Health Center\n",
    "    + Methods: Fitting locally weighted regression model for age vs. height, then estimate confidence intervals (CI) at given age\n",
    "\n",
    "\n",
    "- Semi-supervised encoding for outlier detection in clinical observation data\n",
    "    + Shawn Murphy @ HMS/MGH\n",
    "    + Train autoencoder on samples of vitals/labs/demographics data. The data points with large reconstruction error can be regarded as outliers\n",
    "\n",
    "\n",
    "- City-wide opioid use among patients treated at multiple health care facilities\n",
    "    + Northwestern Medicine\n",
    "    + Data: HealthLNK (linked EMR from multiple medical centers in Chicago)\n",
    "    + Results: \n",
    "        - ~51% of primary care patients (N=884K) had opioid records\n",
    "        - 50% of opioid users visited >=2 institutions, 7% visited >=3, 0.6% visited >=4\n",
    "        - pain-associated complaints are 3-fold more frequent for opioid users who visited more than 1 institutions compared to opioid users who visited only one institution\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# New EMR resources\n",
    "---\n",
    "\n",
    "- MIMIC-III, a freely accessible critical care database, [Scientific Data volume 3, Article number: 160035 (2016)](https://www.nature.com/articles/sdata201635)\n",
    "\n",
    "![](https://media.nature.com/lw926/nature-assets/sdata/2016/sdata201635/images_hires/sdata201635-f1.jpg)\n",
    ">MIMIC-III integrates deidentified, comprehensive clinical data of patients admitted to the Beth Israel Deaconess Medical Center in Boston, Massachusetts    \n",
    "\n",
    ">MIMIC-III contains data associated with **53,423 distinct hospital admissions** for adult patients (aged 16 years or above) admitted to critical care units between 2001 and 2012. In addition, it contains data for **7870 neonates** admitted between 2001 and 2008. The data covers **38,597 distinct adult patients** and 49,785 hospital admissions.\n",
    "\n",
    "- [HealthLNK](http://www.healthlnk.org/)\n",
    "\n",
    "> HealthLNK is an assembly of electronic health records from seven health care institutions: five large Academic Medical Centers (Northwestern Medicine, University of Chicago Hospitals and Clinic, Rush University Medical Center, University of Illinois at Chicago Medical Center, and Loyola University Medical Center); one large county health care system (Cook County Health and Hospital Systems); and a network of community health centers with multiple care outlets (the Alliance of Chicago). The data collection is done under an approved IRB from 2006 â€“ 2012 (inclusive) and resides in an enterprise data warehouse behind one\n",
    "of the institutional firewall.\n",
    "\n",
    "- [IBM Explorys data](https://www-01.ibm.com/common/ssi/cgi-bin/ssialias?htmlfid=GBW03370USEN&) (the site is down...)\n",
    "\n",
    "> - Includes more than 55 million\n",
    "de-identified patients based in the\n",
    "United States and billions of data\n",
    "points that are updated weekly for\n",
    "near-real-time analysis.    \n",
    "> - Provides longitudinal data for these\n",
    "patients, including demographic,\n",
    "diagnostic, and therapeutic information,\n",
    "creating a detailed clinical picture.\n",
    "Many patients are also linked with\n",
    "financial records.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Informatics Implementation Year in Review: So if you don't know, now you know\n",
    "## Leslie McIntosh, Umit Topaloglu\n",
    "\n",
    "\n",
    "### Standards for data exchange \n",
    "\n",
    "\n",
    "FHIR (Fast Healthcare Interoperability Resources): a standard for exchanging healthcare information electronically\n",
    "\n",
    "[c3-pro](http://c3-pro.org/): consent, contact and community framework for patient reported outcomes \n",
    "\n",
    "![](http://c3-pro.org/images/Architecture@2x.png)\n",
    "\n",
    "### AI in clinical informatics\n",
    "\n",
    "[Algorithmia](https://algorithmia.com/): microservice for AI\n",
    "\n",
    "Identify patients for clinical trails:\n",
    "- [Deep 6 AI](https://deep6.ai/)\n",
    "- [Matchminer](https://matchminer.org/): matching patient-specific genomic profiles to precision cancer medicine clinical trials\n",
    "![](https://oncoblocks.files.wordpress.com/2017/03/screenshot_1612.png)\n",
    "\n",
    "Imaging:\n",
    "- [Open Health Imaging Foundation](http://ohif.org/)\n",
    "\n",
    "\n",
    "[Apple Health Records](https://www.apple.com/healthcare/health-records/)\n",
    "\n",
    "\n",
    "### Policies and concepts\n",
    "\n",
    "- General Data Protection Regulation (GDPR): Europe, IP address and cookies\n",
    "- Blockchain (Distributed Ledger Technology): \n",
    "\t[Blockchain improvements to Medical Record](https://academic.oup.com/jamia/article/24/6/1211/4108087)\n",
    "\n",
    "\n",
    "--- \n",
    "\n",
    "\n",
    "\n",
    "# [Beacon Network](https://beacon-network.org/#/): A global search engine for genetic mutations.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Blockchain in Healthcare and Biomedical Research\n",
    "## IBM Research\n",
    "\n",
    "demo: https://git.ng.bluemix.net/IBM-Research-Cambridge-Lab/CHER/Blockchain/amia-demo\n",
    "\n",
    "Concerns about trust and data sharing:\n",
    "- data breaches of the hospitals.\n",
    "- Drs submitting false claims\n",
    "- Medical devices can be hacked\n",
    "- Frauds in clinical trials and medical researches\n",
    "\n",
    "data access control\n",
    "data transactions are transparent\n",
    "data management distributed\n",
    "data provenance\n",
    "\n",
    "When to use Blockchain?\n",
    "- Trusted Third Party?\n",
    "- 3 flavors of Blockchain\n",
    "- milliseconds transactions?\n",
    "\n",
    "Types\n",
    "- Permission Blockchain\n",
    "- Public Blockchain\n",
    "\n",
    "Potential applications \n",
    "- Health info exchange\n",
    "- public health info\n",
    "- research/clinical trails data\n",
    "- provider/patient identities\n",
    "- patient-generated data\n",
    "\n",
    "### Examples\n",
    "\n",
    "- MedRec\n",
    "\t+ Patient App\n",
    "\t+ Provider App \n",
    "\n",
    "- Dynamic Consent Management for Clinical Research\n",
    "\t+ informed consent\n",
    "\t+ 2nd data sharing\n",
    "\n",
    "- Clinical Trials\n",
    "\t+ lock protocol\n",
    "\t+ patients/IRB/researchers\n",
    "\n",
    "### Patient-centered data exchange\n",
    "\n",
    "### Popular Blockchain Frameworks\n",
    "\n",
    "- Crypto currency\n",
    "- social media platforms\n",
    "\n",
    "### Types \n",
    "\n",
    "- Public\n",
    "\t+ anyone can join\n",
    "\t+ proof-of-work (PoW)\n",
    "\t+ incentive for participate\n",
    "\t+ Bitcoin\n",
    "\n",
    "- Permissioned\n",
    "\t+ only known participants\n",
    "\t+ faster and cheaper\n",
    "\t+ smart contracts -> high security\n",
    "\t+ Hyperledger\n",
    "\n",
    "both distributed \n",
    "\n",
    "### Bitcoin\n",
    "2008\n",
    "**immutable** transactions \n",
    "a block:\n",
    "- Data (transactions)\n",
    "- Hash\n",
    "- Hash of previous block\n",
    "\n",
    "PoW:\n",
    "Transaction + Nonce -> SHA-256 -> hash starts with 'n' zeros\n",
    "\n",
    "### Ethereum\n",
    "\n",
    "both public and private\n",
    "PoW\n",
    "smart contract\n",
    "\n",
    "### Hyperledger Fabric (HF)\n",
    "only private network\n",
    "appropriate for healthcare\n",
    "\n",
    "components:\n",
    "- network \n",
    "\t+ client: propose transactions\n",
    "\t+ peer: commit transactions\n",
    "\t+ ordered node: creates orders of transactions\n",
    "\n",
    "- ledger (stores transactions and data)\n",
    "\t+ state DB: key-val pairs, only smart contract access data from DB\n",
    "\t+ transactions blocks: stores transactions\n",
    "- smart contract (chaincode)\n",
    "\t+ read/write through its functions\n",
    "\n",
    "- Endorsement policy: who in the network is allowed to endorse transactions\n",
    "- Application interface\n",
    "\n",
    "features:\n",
    "- private channels\n",
    "- fine-grained access control: e.g. access data through a certain time frame\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
